api/src/test/java/org/openmrs/api/ProviderServiceTest.java

```diff
@@ -545,5 +587,119 @@ public void getUnknownProvider_shouldGetTheUnknownProviderAccount() {
+	 */
+	@Test
+	public void getProvider_shouldReturnProviderWithAssociatedProviderRole() {
+		Provider provider = service.getProvider(1009);
+		assertNotNull(provider);
+		assertEquals(1005, provider.getProviderRole().getProviderRoleId());
+		assertEquals("Community health nurse", provider.getProviderRole().getName());
+		assertEquals("da7f623f-37ce-4bb2-86d6-6d1d05312bd5", provider.getProviderRole().getUuid());
+	}
+
+	@Test
+	public void getAllProviderRoles_shouldGetAllProviderRoles() {
+		List<ProviderRole> roles = service.getAllProviderRoles(true);
+		int roleCount = roles.size();
+		assertEquals(12, roleCount);
+
+		roles = service.getAllProviderRoles(true);
+		roleCount = roles.size();
+		assertEquals(12, roleCount);
+	}
+
+	@Test
+	public void getAllProviderRoles_shouldGetAllProviderRolesExcludingRetired() {
+		List<ProviderRole> roles = service.getAllProviderRoles(false);
+		int roleCount = roles.size();
+		assertEquals(11, roleCount);
+	}
+
+	@Test
+	public void getProviderRole_shouldGetProviderRole() {
+		ProviderRole role = service.getProviderRole(1002);
+		assertEquals(1002, role.getId());
+		assertEquals("Community Health Worker", role.getName());
+	}
+
+	@Test
+	public void getProviderRole_shouldReturnNullIfNoProviderForId() {
+		assertNull(service.getProviderRole(200));
+	}
+
+	@Test
+	public void getProviderRoleByUuid_shouldGetProviderRoleByUuid() {
+		ProviderRole role = service.getProviderRoleByUuid("db7f523f-27ce-4bb2-86d6-6d1d05312bd5");
+		assertEquals(1003, role.getId());
+		assertEquals("Cell supervisor", role.getName());
+	}
+
+	@Test
+	public void getProviderRoleByUuid_shouldReturnNullIfNoProviderForUuid() {
+		ProviderRole role = service.getProviderRoleByUuid("zzz");
+	}
+
+	@Test
+	public void saveProviderRole_shouldSaveProviderRole() {
+		ProviderRole role = new ProviderRole();
+		role.setName("Some provider role");
+
+		ProviderRole saved = Context.getService(ProviderService.class).saveProviderRole(role);
+
+		assertNotNull(saved.getProviderRoleId());
+		assertEquals("Some provider role", saved.getName());
+
+		ProviderRole fetched = service.getProviderRole(saved.getProviderRoleId());
+		assertNotNull(fetched);
+		assertEquals("Some provider role", fetched.getName());
+	}
+
+	@Test
+	public void deleteProviderRole_shouldDeleteProviderRole() throws Exception {
+		ProviderRole role = service.getProviderRole(1012);
+		assertNotNull(role);
+
+		service.purgeProviderRole(role);
+
+		ProviderRole deleted = service.getProviderRole(1012);
+		assertNull(deleted);
+	}
+
+	@Test
+	public void retireProviderRole_shouldRetireProviderRole() {
+		ProviderRole role = service.getProviderRole(1002);
+		assertFalse(role.getRetired());
+
+		service.retireProviderRole(role, "test reason");
+
+		ProviderRole retiredRole = service.getProviderRole(1002);
+		assertTrue(retiredRole.getRetired());
+		assertEquals("test reason", retiredRole.getRetireReason());
+
+		List<ProviderRole> activeRoles = service.getAllProviderRoles(false);
```

Comment:
This is not necessary after the above. But you want to keep it, then you need to assert that it originally contained it.